### ISSUE TITLE
Gunicorn uses 100% CPU constantly

### ISSUE THREAD

**Issue by reinaertvdc** [CTFd/CTFd#849]
**Environment**:

  - CTFd Version/Commit: 2.0.3
  - Operating System: Ubuntu 18.04
  - Web Browser and Version: N/A for this issue (but I use Google Chrome 71)

**What happened?**

I use stock dockerized CTFd. In this setup, Gunicorn uses 100% CPU constantly, without interruption, for a straight week now, without any clients. On reboot, the CPU stays idle for 30 minutes, after which Gunicorn causes it to jump to a permanent 100% again. The setup runs on the smallest DigitalOcean droplet (1 vCPU, 1 GB RAM, Ubuntu 18.04). Resetting and/or exporting and importing CTFd have no effect on CPU usage whatsoever.

Concerning RAM usage, DigitalOcean monitoring reports that Gunicorn uses ~327MB RAM, and docker stats reports that the ctfd container uses ~268MB RAM (and ~95% CPU). The db and cache containers meanwhile are idle (with ~93MB and ~2MB RAM usage respectively). 

**What did you expect to happen?**

I expected Gunicorn to use much less CPU when there are no clients.

**How to reproduce your issue**

It's just a stock dockerized CTFd setup on the smallest DigitalOcean droplet. The only changes I made myself were to add a few challenges and a few admin users. There aren't any participants or teams yet.

**Any associated stack traces or error logs**

The CTFd logs directory is empty. I will provide any data required if I get a description of how to produce it.

**Comment by ColdHeat** [comment]
I've seen some similar issues but it's quite difficult for me to reproduce. But generally if you kill the worker, the new one will not exhibit the same issue. 

Some things that could be helpful here. Could you try to disable Flask SocketIO on your installation? You'd comment out these lines: https://github.com/CTFd/CTFd/blob/master/CTFd/__init__.py#L151-L155

I can help you dig into this on the CTFd Slack if you'd like: https://slack.ctfd.io/

**Comment by reinaertvdc** [comment]
Thanks for your quick response! I'll pick up on this tomorrow, it's late now in my timezone.

**Comment by reinaertvdc** [comment]
The latest commit seems to have fixed the issue: https://github.com/CTFd/CTFd/commit/ae8ce0b4309fa62d5781b26754c6f7463c32e584

For reference, I found that the CPU spikes were caused by opening Chrome Developer Tools. As soon as I loaded a new page with devtools opened, gunicorn would spike to 100% CPU and stay there indefinitely. Restart containers, everything remains fine until I opened devtools again. Now, with the latest commit, I don't have this problem anymore.
